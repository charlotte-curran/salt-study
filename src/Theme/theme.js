export const SaltTheme = {
  color: {
    dark: "#222324",
    blue: "#92b3d2",
    white: "rgba(255,255,255, 1)",
    orange: "#ff7a61",
    shadow: "#545353",
    address: "rgba(255, 255, 255, 0.4)",
    navbar: "rgba(34, 35, 36, 0.7)"
  },
  fonts: {
    futuraPt: '"futura-pt", sans-serif',
    raleway: '"raleway", sans-serif'
  }
};
