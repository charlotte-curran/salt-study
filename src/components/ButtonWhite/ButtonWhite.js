import React from "react";
import { Button } from "./ButtonWhite.styles";

const ButtonWhite = props => {
  const { isFooter } = props;
  return (
    <Button isFooter={isFooter} onClick={props.onClick}>
      {props.children}
    </Button>
  );
};

export default ButtonWhite;
