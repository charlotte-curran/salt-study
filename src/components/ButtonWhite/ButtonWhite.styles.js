import styled from "styled-components";

export const Button = styled.button`
  background-color: ${({ theme }) => theme.color.dark};
  color: ${({ theme }) => theme.color.blue};
  cursor: pointer;
  box-shadow: ${({ theme }) => `0px 1px 4px ${theme.color.shadow}`};
  transition: all 300ms ease;
  font-family: ${({ theme }) => theme.fonts.raleway};
  font-size: 11px;
  text-transform: uppercase;
  letter-spacing: 0.3rem;
  padding-left: 15px;
  padding-right: 15px;
  border: ${({ theme }) => `outset 2px ${theme.color.white}`};
  height: 35px;
  letter-spacing: 3px;

  :hover {
    box-shadow: ${({ theme, isFooter }) =>
      !isFooter
        ? `0px 1px 10px ${theme.color.shadow}`
        : `0px 1px 10px ${theme.color.white}`};
  }

  @media (min-width: 1000px) {
    font-size: 15px;
    height: 42px;
    padding-left: 28px;
    padding-right: 28px;
  }
`;
