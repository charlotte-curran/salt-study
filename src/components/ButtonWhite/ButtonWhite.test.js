import React from "react";
import { create } from "react-test-renderer";
import ButtonWhite from "./ButtonWhite";
import { ThemeProvider } from "styled-components";
import { SaltTheme } from "../../Theme/theme";

describe("ButtonWhite", () => {
  test("matches the snapshot", () => {
    const buttonWhite = create(
      <ThemeProvider theme={SaltTheme}>
        <ButtonWhite isFooter={false} />
      </ThemeProvider>
    );
    expect(buttonWhite.toJSON()).toMatchSnapshot();
  });
});
