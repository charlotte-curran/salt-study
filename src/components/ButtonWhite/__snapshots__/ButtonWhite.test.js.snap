// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonWhite matches the snapshot 1`] = `
<button
  className="sc-bdVaJa iGLASH"
/>
`;
